### PR TITLE
Map some poi and village stuff

### PR DESCRIPTION
**Pull request by liach**
Signed-off-by: liach <liach@users.noreply.github.com>

### CODE REVIEW

File: mappings/net/minecraft/container/MerchantContainer.mapping [hunk 1]
```diff
@@ -1,22 +1,31 @@
 CLASS ayy net/minecraft/container/MerchantContainer
-	FIELD c villager Lbgd;
-	FIELD d villagerInventory Layx;
+	FIELD c trader Lbgd;
+	FIELD d traderInventory Layx;
```
Comment (Runemoro) [inline mappings/net/minecraft/container/MerchantContainer.mapping:3]: Why do we have `MerchantContainer` but `traderInventory`? ~~I think we should either use "merchant" or "trader". Do we know which one is the Mojang name?~~ Mojang calls them "traders", I think `MerchantContainer` should be renamed to `TraderContainer`, or even better, `TraderInventory`.

File: mappings/net/minecraft/container/MerchantContainer.mapping [hunk 2]
```diff
@@ -1,22 +1,31 @@
...
 		ARG 1 syncId
+		ARG 2 playerInventory
+		ARG 3 trader
 	METHOD a setRecipes (Lbgf;)V
 	METHOD a setCanLevel (Z)V
 		ARG 1 canLevel
 	METHOD b equals (Lbbu;Lbbu;)Z
 		ARG 1 itemStack
 		ARG 2 otherItemStack
+	METHOD c pullMatchingStackFromPlayerInventory (ILbbu;)V
```
Comment (Runemoro) [inline mappings/net/minecraft/container/MerchantContainer.mapping]: Why not just `takeFromPlayer`?

File: mappings/net/minecraft/container/TradeOutputSlot.mapping [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+	FIELD b player Lavr;
+	FIELD g amount I
+	FIELD h trader Lbgd;
+	METHOD <init> (Lavr;Lbgd;Layx;III)V
+		ARG 1 player
+		ARG 2 trader
+		ARG 3 traderInventory
+		ARG 4 index
+		ARG 5 xPosition
+		ARG 6 yPosition
```
Comment (Runemoro) [inline mappings/net/minecraft/container/TradeOutputSlot.mapping]: I think just `x` and `y` would be enough.

File: mappings/net/minecraft/entity/ai/brain/sensor/SecondaryJobSiteSensor.mapping [hunk 4]
```diff
@@ -0,0 +1 @@
+CLASS apm net/minecraft/entity/ai/brain/sensor/SecondaryJobSiteSensor
```
Comment (Runemoro) [inline mappings/net/minecraft/entity/ai/brain/sensor/SecondaryJobSiteSensor.mapping]: Why was this renamed? Look at the ID in `SensorType`:
```java
SECONDARY_POIS = SensorType.register("secondary_pois", SecondaryPointsOfInterestSensor::new);
```

File: mappings/net/minecraft/village/PointOfInterestStorage.mapping [hunk 5]
```diff
@@ -9,6 +9,9 @@ CLASS aqa net/minecraft/village/PointOfInterestStorage
 		FIELD d predicate Ljava/util/function/Predicate;
 		METHOD a getPredicate ()Ljava/util/function/Predicate;
 	FIELD a pointOfInterestDistanceTracker Laqa$a;
+	METHOD a initForPalette (Lbgn;Lbxd;)V
+	METHOD a mayPointOfInterestAppearIn (Lbxd;)Z
+	METHOD a createPointsOfInterest (Lbxd;Lfo;Ljava/util/function/BiConsumer;)V
```
Comment (Runemoro) [inline mappings/net/minecraft/village/PointOfInterestStorage.mapping]: I think these shouldn't have `PointOfInterest` in the names to match the other methods (`add`, `remove`, `getDistanceFromNearestOccupied`, `count`). 

Also, what does this method do differently from `add`?

Comment (liach) [reply mappings/net/minecraft/village/PointOfInterestStorage.mapping]: this one creates poises if they aren't present; then it adds poi for each matching state in the chunk

File: mappings/net/minecraft/container/MerchantContainer.mapping (hunk 1 above)
Comment (liach) [reply mappings/net/minecraft/container/MerchantContainer.mapping:3]: mojabg calls the gui merchant container.
btw, where does mojang call trader?

Comment (Runemoro) [reply mappings/net/minecraft/container/MerchantContainer.mapping:3]: Weird, the ID for the trader is `wandering_trader`, but the container is `merchant` rather than `trader`...

Comment (Prospector) [reply mappings/net/minecraft/container/MerchantContainer.mapping:3]: villagers are merchants too. The UI is a merchant UI and the entity is a trader. It makes sense to me because merchants don't necessarily have to be traders, even if that's all vanilla ones are at the moment.

File: mappings/net/minecraft/container/MerchantContainer.mapping (hunk 2 above)
Comment (liach) [reply mappings/net/minecraft/container/MerchantContainer.mapping]: It is autofill, like what recipe books do.

File: mappings/net/minecraft/container/MerchantContainer.mapping (hunk 1 above)
Comment (liach) [reply mappings/net/minecraft/container/MerchantContainer.mapping:3]: so `merchant` for all recipes, ui, etc.?

File: mappings/net/minecraft/container/MerchantContainer.mapping (hunk 2 above)
Comment (Runemoro) [reply mappings/net/minecraft/container/MerchantContainer.mapping]: But isn't there a shorter name? At least `Inventory` is unnecessary.

File: mappings/net/minecraft/entity/ai/brain/sensor/SecondaryJobSiteSensor.mapping (hunk 4 above)
Comment (liach) [reply mappings/net/minecraft/entity/ai/brain/sensor/SecondaryJobSiteSensor.mapping]: i saw it was using secondary job site from memory module types

**Review by Runemoro — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]
